src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,299 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/decay/interactor/MuDecayInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Constants.hh"
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/data/StackAllocator.hh"
+#include "corecel/math/Algorithms.hh"
+#include "corecel/math/ArrayUtils.hh"
+#include "corecel/math/EulerRotation.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/decay/data/MuDecayData.hh"
+#include "celeritas/phys/FourVector.hh"
+#include "celeritas/phys/Interaction.hh"
+#include "celeritas/phys/InteractionUtils.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/phys/Secondary.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform muon decay.
+ *
+ * Only one decay channel is implemented, with muons decaying to
+ * \f[
+ * \mu^- \longrightarrow e^- \overline{\nu}_e \nu_\mu
+ * \f]
+ * or
+ * \f[
+ * \mu^+ \longrightarrow e^+ \nu_e \overline{\nu}_\mu
+ * \f].
+ *
+ * This interactor follows \c G4MuonDecayChannel and the Physics Reference
+ * Manual, Release 11.2, section 4.2.3.
+ */
+class MuDecayInteractor
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Energy = units::MevEnergy;
+    //!@}
+
+    // Construct with shared and state data
+    inline CELER_FUNCTION
+    MuDecayInteractor(MuDecayData const& shared,
+                      ParticleTrackView const& particle,
+                      Real3 const& inc_direction,
+                      StackAllocator<Secondary>& allocate);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// DATA ////
+
+    // Constant data
+    MuDecayData shared_;
+    // Incident muon energy
+    units::MevEnergy const inc_energy_;
+    // Incident muon direction
+    Real3 const inc_direction_;
+    // Allocate space for a secondary particles
+    StackAllocator<Secondary>& allocate_;
+    // Define decay channel based on muon or anti-muon primary
+    ParticleId secondary_id_[3];
```

Comment:
I initially had a `is_muon_` boolean assigned via the `ParticlTrackView`, but ended up with that array so that I could set up the particle decay channel during construction instead of operator().

Since we are agreeing on allocating the electron only, I'll revert to the boolean, which is enough to define the secondary id. This will also clean up the neutrino ids in `MuDecayData` and `PDGNumber`...